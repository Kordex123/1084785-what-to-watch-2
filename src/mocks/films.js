const films = [
  {
    id: 10111,
    genre: `Comedies`,
    starring: [],
    movieTitle: `Fantastic Beasts: The Crimes of Grindelwald`,
    previewMovieImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideoLink: `http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerMeltdowns.mp4`,
  },
  {
    id: 20222,
    genre: `Comedies`,
    movieTitle: `Bohemian Rhapsody`,
    previewMovieImage: `bohemian-rhapsody.jpg`,
    previewVideoLink: `http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerFun.mp4`,
  },
  {
    id: 30333,
    genre: `Documentary`,
    movieTitle: `Macbeth`,
    previewMovieImage: `macbeth.jpg`,
    previewVideoLink: `https://file-examples.com/wp-content/uploads/2017/04/file_example_MP4_640_3MG.mp4`,
  },
  {
    id: 40444,
    genre: `Romance`,
    movieTitle: `Aviator`,
    previewMovieImage: `aviator.jpg`,
    previewVideoLink: `https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4
`,
  },
  {
    id: 50555,
    genre: `Romance`,
    movieTitle: `Pulp Fiction`,
    previewMovieImage: `pulp-fiction.jpg`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

export default films;
