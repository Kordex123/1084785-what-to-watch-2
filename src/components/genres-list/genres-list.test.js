import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import GenresList from "./genres-list.jsx";
import {Provider} from 'react-redux';
import Namespace from "../../reducer/namespace";
import films from "../../mocks/films";

it(`GenreList correctly renders after relaunch`, () => {
  const mockStore = configureStore([]);
  let store = mockStore({
    [Namespace.MOVIE]: {
      movieCards: films,
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <GenresList />
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
