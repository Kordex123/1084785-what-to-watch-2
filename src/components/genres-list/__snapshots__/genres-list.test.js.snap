// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList correctly renders after relaunch 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Comedies
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Romance
    </a>
  </li>
</ul>
`;
